src/utils/index.js

```diff
@@ -88,6 +88,30 @@ function getAllTemplates() {
   return templates;
 }
 
+function getTemplatesByCategory() {
+  const templatePaths = fs
+    .readdirSync(TEMPLATES_FOLDER)
+    .map(name => path.join(TEMPLATES_FOLDER, name))
+    .filter(source => fs.lstatSync(source).isDirectory());
+
+  const templates = templatePaths.reduce((allTemplates, source) => {
+    const name = path.basename(source);
+
+    const { category } = require(`${source}/.template.js`);
+
+    if (!category) {
+      return allTemplates;
+    }
+
+    const newAllTemplates = allTemplates;
```

Comment:
The temporary variable is useless I think, we modify the reference of the object on the next line. We can inline the return statement with a spread operator.

Reply:
We've got an ESLint rule `no-param-reassign` that disallows to modify the reference object. I think it's a lot more work to inline the return statement with a spread operator in this case. I might be wrong?

Reply:
Like you want, I personally prefer to avoid mutations on objects but it's my point of view.

Reply:
So do I. However I had the feeling it might be easier to understand here. I'll compare both versions!

Reply:
It's fine actually:

```javascript
{
  ...allTemplates,
  [category]: [...(allTemplates[category] || []), name],
}
```